CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+CMAKE_MINIMUM_REQUIRED(VERSION 2.8.12)
+
+FIND_PACKAGE(deal.II 9.3.0 QUIET
+  HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
+  )
+IF(NOT ${deal.II_FOUND})
+  MESSAGE(FATAL_ERROR "\n"
+    "*** Could not locate deal.II. ***\n\n"
+    "You may want to either pass a flag -DDEAL_II_DIR=/path/to/deal.II to cmake\n"
+    "or set an environment variable \"DEAL_II_DIR\" that contains this path."
+    )
+ENDIF()
+
+IF(NOT DEAL_II_WITH_P4EST) # keep in one line
+  MESSAGE(FATAL_ERROR "
+Error! The deal.II library found at ${DEAL_II_PATH} was not configured with
+    DEAL_II_WITH_P4EST = ON
+It was set to OFF in your installation but is required for this program."
+    )
+ENDIF()
+
+DEAL_II_INITIALIZE_CACHED_VARIABLES()
+
+SET( TARGET_SRC dummy.cc)
+
+PROJECT(multigrid)
+
+ADD_LIBRARY(multigrid  ${TARGET_SRC})
+
+# Define custom targets to easily switch the build type:
+ADD_CUSTOM_TARGET(debug
+  COMMAND ${CMAKE_COMMAND} -DCMAKE_BUILD_TYPE=Debug ${CMAKE_SOURCE_DIR}
+  COMMENT "Switch CMAKE_BUILD_TYPE to Debug\nPlease call 'make' to build the debug project"
+  )
+
+ADD_CUSTOM_TARGET(release
+  COMMAND ${CMAKE_COMMAND} -DCMAKE_BUILD_TYPE=Release ${CMAKE_SOURCE_DIR}
+  COMMENT "Switch CMAKE_BUILD_TYPE to Release\nPlease call 'make' to build the release project"
+  )
+
+DEAL_II_INITIALIZE_CACHED_VARIABLES()
+
+DEAL_II_SETUP_TARGET(multigrid)
+
+IF(EXISTS ${CMAKE_SOURCE_DIR}/tests/CMakeLists.txt)
+  SET(DEAL_II_HAVE_TESTS_DIRECTORY TRUE)
+ENDIF()
+
+enable_testing()
+
+ADD_SUBDIRECTORY(cmake)
+
+ADD_SUBDIRECTORY(matvec_dg)
+ADD_SUBDIRECTORY(matvec_dg_cheby)
+#ADD_SUBDIRECTORY(minimal_surface)
+ADD_SUBDIRECTORY(poisson_cube)
+ADD_SUBDIRECTORY(poisson_dg)
+ADD_SUBDIRECTORY(poisson_dg_plain)
+#ADD_SUBDIRECTORY(poisson_l)
```

Comment:
I was not able to get these two programs running.

Reply:
They should be running - I will need to have a look.

Reply:
I need to postpone it to #3.